src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -159,7 +159,7 @@ private static string EmitResponseWritingCall(this Endpoint endpoint)
     public static string EmitFilteredRequestHandler(this Endpoint endpoint, int baseIndent = 0)
     {
         var argumentList = endpoint.Parameters.Length == 0 ? string.Empty : $", {endpoint.EmitArgumentList()}";
-        var invocationConstructor = endpoint.Parameters.Length == 0 ? "DefaultEndpointFilterInvocationContext" : "EndpointFilterInvocationContext";
+        var invocationConstructor = endpoint.Parameters.Length == 0 || endpoint.Parameters.Length > 8 ? "new DefaultEndpointFilterInvocationContext" : "EndpointFilterInvocationContext.Create";
         var invocationGenericArgs = endpoint.Parameters.Length == 0 ? string.Empty : $"<{endpoint.EmitFilterInvocationContextTypeArgs()}>";
```

Comment:
Does this still work correctly when `endpoint.Parameters.Length` is greater than 8?

Maybe we need a test for `> 8`( and  possibly `> 10` for the internal code?).

Reply:
We've got existing tests in the RDF that cover this. I went ahead and moved them to the shared tests and added a few more scenarios so we have coverage for both targets here.